include/zephyr/drivers/bluetooth.h

```diff
@@ -0,0 +1,253 @@
+ * @brief Bluetooth HCI APIs
+ * @defgroup bt_hci_api Bluetooth HCI APIs
+ * @ingroup bluetooth
+ * @{
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/buf.h>
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/bluetooth/hci_vs.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct bt_hci_setup_params {
+	/** The public identity address to give to the controller. This field is used when the
+	 *  driver selects @kconfig{CONFIG_BT_HCI_SET_PUBLIC_ADDR} to indicate that it supports
+	 *  setting the controller's public address.
+	 */
+	bt_addr_t public_addr;
+};
+
+enum {
+	/* The host should never send HCI_Reset */
+	BT_HCI_QUIRK_NO_RESET = BIT(0),
+	/* The controller does not auto-initiate a DLE procedure when the
+	 * initial connection data length parameters are not equal to the
+	 * default data length parameters. Therefore the host should initiate
+	 * the DLE procedure after connection establishment.
+	 */
+	BT_HCI_QUIRK_NO_AUTO_DLE = BIT(1),
+};
+
+/** Possible values for the 'bus' member of the bt_hci_driver struct */
+enum bt_hci_bus {
+	BT_HCI_BUS_VIRTUAL       = 0,
+	BT_HCI_BUS_USB           = 1,
+	BT_HCI_BUS_PCCARD        = 2,
+	BT_HCI_BUS_UART          = 3,
+	BT_HCI_BUS_RS232         = 4,
+	BT_HCI_BUS_PCI           = 5,
+	BT_HCI_BUS_SDIO          = 6,
+	BT_HCI_BUS_SPI           = 7,
+	BT_HCI_BUS_I2C           = 8,
+	BT_HCI_BUS_IPM           = 9,
+};
+
+#define BT_DT_HCI_QUIRK_OR(node_id, prop, idx) DT_STRING_TOKEN_BY_IDX(node_id, prop, idx)
+#define BT_DT_HCI_QUIRKS_GET(node_id) COND_CODE_1(DT_NODE_HAS_PROP(node_id, bt_hci_quirks), \
+						  (DT_FOREACH_PROP_ELEM_SEP(node_id, \
+									    bt_hci_quirks, \
+									    BT_DT_HCI_QUIRK_OR, \
+									    (|))), \
+						  (0))
+#define BT_DT_HCI_QUIRKS_INST_GET(inst) BT_DT_HCI_QUIRKS_GET(DT_DRV_INST(inst))
+
+#define BT_DT_HCI_NAME_GET(node_id) DT_PROP_OR(node_id, bt_hci_name, "HCI")
+#define BT_DT_HCI_NAME_INST_GET(inst) BT_DT_HCI_NAME_GET(DT_DRV_INST(inst))
+
+#define BT_DT_HCI_BUS_GET(node_id) DT_STRING_TOKEN_OR(node_id, bt_hci_bus, BT_HCI_BUS_VIRTUAL)
+#define BT_DT_HCI_BUS_INST_GET(inst) BT_DT_HCI_BUS_GET(DT_DRV_INST(inst))
+
+typedef int (*bt_hci_recv_t)(const struct device *dev, struct net_buf *buf);
+
+__subsystem struct bt_hci_driver_api {
+	int (*open)(const struct device *dev, bt_hci_recv_t recv);
+	int (*close)(const struct device *dev);
+	int (*send)(const struct device *dev, struct net_buf *buf);
+#if defined(CONFIG_BT_HCI_SETUP)
+	int (*setup)(const struct device *dev,
+		     const struct bt_hci_setup_params *param);
+#endif /* defined(CONFIG_BT_HCI_SETUP) */
+};
+
+/**
+ * @brief Open the HCI transport.
+ *
+ * Opens the HCI transport for operation. This function must not
+ * return until the transport is ready for operation, meaning it
+ * is safe to start calling the send() handler.
+ *
+ * @param dev  HCI device
+ * @param recv Callback to receive data from the controller
```

Comment:
Can we document in which execution context `bt_hci_recv_t recv` is to be called?

Reply:
> Can we document in which execution context `bt_hci_recv_t recv` is to be called?

Yes, added.